### PR TITLE
A new class, Player, to replay a journal

### PR DESCRIPTION
**Pull request by pleroy**

### CODE REVIEW

File: ksp_plugin/journal.cpp [hunk 1]
```diff
@@ -3,20 +3,50 @@
...
+
+// Recursively reads a line of arbitrary length.
+std::string GetLine(not_null<std::ifstream*> const stream) {
+  char buffer[kBufferSize];
+  if (!stream->getline(&buffer[0], kBufferSize).eof() && stream->fail()) {
+    stream->clear();
+    return buffer + GetLine(stream);
+  }
+  return buffer;
+}
```
Comment (eggrobin) [inline ksp_plugin/journal.cpp]: Move to `base`.

File: ksp_plugin/journal.cpp [hunk 2]
```diff
@@ -3,20 +3,50 @@
...
+template<typename T>
+void Insert(not_null<PointerMap*> const pointer_map,
+            std::uint64_t const address,
+            T* const pointer) {
+  auto inserted = pointer_map->emplace(address, pointer);
+  CHECK(inserted.second) << address;
+}
+
+// Recursively reads a line of arbitrary length.
+std::string GetLine(not_null<std::ifstream*> const stream) {
```
Comment (eggrobin) [inline ksp_plugin/journal.cpp]: Comment that this implementation is inefficient.

File: ksp_plugin/journal.cpp (hunk 1 above)
Comment (pleroy) [inline ksp_plugin/journal.cpp]: Done.

File: ksp_plugin/journal.cpp (hunk 2 above)
Comment (pleroy) [inline ksp_plugin/journal.cpp]: Done.

**Pull request merged**